### ISSUE TITLE
Model weights contain NaN/Inf values - unable to reproduce paper results

### ISSUE THREAD

**Issue by idaka9527** [liufanfanlff/C3-Context-Cascade-Compression#4]
### Problem Description

     I am trying to reproduce the text reconstruction results from the paper, but the model
      outputs meaningless sequences (`!!!!!...`) instead of reconstructing the input text.
     After detailed investigation, I found that **the model weights downloaded from 
     HuggingFace contain NaN and Inf values**, which causes the forward pass to produce NaN
      logits.

     ### Environment

     - **OS**: Ubuntu Linux
     - **Python**: 3.10
     - **PyTorch**: 2.6.0+cu124
     - **Transformers**: 4.49.0
     - **GPU**: NVIDIA RTX 4090
     - **CUDA**: 12.4

     ### Steps to Reproduce

     1. Download model from HuggingFace:
     ```python
     from transformers import AutoModel, AutoTokenizer
     import torch

     model_name = 'liufanfanlff/C3-Context-Cascade-Compression'
     tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)
     model = AutoModel.from_pretrained(
         model_name,
         trust_remote_code=True,
         low_cpu_mem_usage=True,
         device_map='cuda',
         torch_dtype=torch.bfloat16,
         use_safetensors=True,
         pad_token_id=tokenizer.eos_token_id
     )
     model = model.eval().cuda()

   2. Run the official example:

   python
     prompt = 'Repeat the text: '
     context = "帝高阳之苗裔兮，朕皇考曰伯庸。摄提贞于孟陬兮，"
     outputs = model.chat(tokenizer, context, prompt)
     print(outputs)

   3. Expected output: The original Chinese text should be reconstructed
   4. Actual output: !!!!!!!!!!!!!!!!!!!!\"!!!!!!!!!!!!!!!!!!!#!!!!!!!!!!!!!!!!!!!$...
      (4096 characters of incrementing token IDs)

   Root Cause Analysis

   I investigated the model weights and found:

   python
     import torch

     # Check for NaN/Inf in model parameters
     nan_params = []
     inf_params = []
     for name, param in model.named_parameters():
         if torch.isnan(param).any():
             nan_params.append(name)
         if torch.isinf(param).any():
             inf_params.append(name)

     print(f"Parameters with NaN: {len(nan_params)}")  # Output: 127
     print(f"Parameters with Inf: {len(inf_params)}")  # Output: 8

   Results:
   •  127 parameters contain NaN values
   •  8 parameters contain Inf values
   •  llm1.model.embed_tokens.weight: all NaN
   •  llm1.lm_head.weight: all NaN
   •  Multiple transformer layers (2, 3, 4, 5, 7, 20, 22, 29, 30, 31, 32) have NaN or Inf

   Affected layers include:

     model.layers.2.self_attn.o_proj.weight (NaN)
     model.layers.2.mlp.gate_proj.weight (NaN)
     model.layers.3.self_attn.v_proj.weight (NaN)
     model.layers.5.mlp.down_proj.weight (Inf)
     model.layers.7.self_attn.q_proj.weight (Inf)
     ... and many more

   When I check the forward pass logits:

   python
     outputs = model(input_ids=input_ids, context_ids=context_ids, return_dict=True)
     print(f"Logits has NaN: {torch.isnan(outputs.logits).any()}")  # True
     print(f"Logits min: {outputs.logits.min()}")  # nan
     print(f"Logits max: {outputs.logits.max()}")  # nan

   Since all logits are NaN, argmax returns index 0 for every position, which decodes to !
   (token ID 0), resulting in the !!!!!... output pattern.

   File Verification

   I verified that the downloaded files match HuggingFace:
   •  model-00001-of-00002.safetensors: 4,957,249,008 bytes ✓
   •  model-00002-of-00002.safetensors: 1,220,760,840 bytes ✓
   •  llm1/model.safetensors: 3,087,467,144 bytes ✓
   •  modeling_C3.py: MD5 matches HuggingFace ✓
   •  config.json: MD5 matches HuggingFace ✓

   Questions

   1. Could you please check if the model weights uploaded to HuggingFace are correct?
   2. Is there a specific checkpoint or version that can reproduce the paper's results
      (98%+ reconstruction precision)?
   3. Are there any additional initialization steps required that are not documented?

   Additional Information

   The paper mentions training for 40,000 steps on 8×H800 GPUs. The _name_or_path in
   llm1/config.json shows /checkpoint-1/llm1, which suggests this might be an early
   checkpoint rather than the final trained model.

   Thank you for your excellent work on C3! Looking forward to your response.
